pkg/arm/resources.go

```diff
@@ -0,0 +1,530 @@
+package arm
+
+import (
+	"encoding/base64"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute"
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-10-01/network"
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2015-06-15/storage"
+	"github.com/Azure/go-autorest/autorest/to"
+
+	"github.com/openshift/openshift-azure/pkg/api"
+	"github.com/openshift/openshift-azure/pkg/config"
+	"github.com/openshift/openshift-azure/pkg/jsonpath"
+	"github.com/openshift/openshift-azure/pkg/tls"
+	"github.com/openshift/openshift-azure/pkg/util/template"
+)
+
+const (
+	vnetName                             = "vnet"
+	vnetSubnetName                       = "default"
+	ipAPIServerName                      = "ip-apiserver"
+	lbAPIServerName                      = "lb-apiserver"
+	lbAPIServerFrontendConfigurationName = "frontend"
+	lbAPIServerBackendPoolName           = "backend"
+	lbAPIServerLoadBalancingRuleName     = "port-443"
+	lbAPIServerProbeName                 = "port-443"
+	nsgMasterName                        = "nsg-master"
+	nsgMasterAllowSSHRuleName            = "allow_ssh"
+	nsgMasterAllowHTTPSRuleName          = "allow_https"
+	nsgWorkerName                        = "nsg-worker"
+	vmssNicName                          = "nic"
+	vmssNicPublicIPConfigurationName     = "ip"
+	vmssIPConfigurationName              = "ipconfig"
+	vmssCSEName                          = "cse"
+	vmssAdminUsername                    = "cloud-user"
+)
+
+// fixupAPIVersions inserts an apiVersion field into the ARM template for each
+// resource (the field is missing from the internal Azure type).  The versions
+// referenced here must be kept in lockstep with the imports above.
+func fixupAPIVersions(template map[string]interface{}) {
+	for _, resource := range jsonpath.MustCompile("$.resources.*").Get(template) {
+		typ := jsonpath.MustCompile("$.type").MustGetString(resource)
+		var apiVersion string
+		switch typ {
+		case "Microsoft.Compute/virtualMachineScaleSets":
+			apiVersion = "2018-06-01"
+		case "Microsoft.Network/loadBalancers",
+			"Microsoft.Network/networkSecurityGroups",
+			"Microsoft.Network/publicIPAddresses",
+			"Microsoft.Network/virtualNetworks":
+			apiVersion = "2017-10-01"
+		case "Microsoft.Storage/storageAccounts":
+			apiVersion = "2015-06-15"
```

Comment:
Question for a different PR but is there any reason not to move to a newer version?